vital/types/geo_offset.h

```diff
@@ -0,0 +1,210 @@
+ */
+class VITAL_EXPORT geo_offset : public point_3d
+{
+public:
+  typedef Eigen::Matrix< double, 3, 1 > vector_type;
+
+  geo_offset()
+  {
+    m_value = vector_type::Zero();
+  }
+  geo_offset(vector_type const& v) { m_value=v; }
+
+  virtual ~geo_offset() = default;
+
+  geo_point_sptr& origin();
+  const geo_point_cptr origin() const;
+
+  /**
+   * \brief Get the WGS84 coordinates of the cartesian coordinates.
+   *
+   * \returns The WGS84 lon/lat/alt of the cartesian coordinates.
+   */
+  vector_type get_lon_lat_alt() const;
+  /**
+   * \brief Set the cartesian coordinates based on 2 geo_points
+   */
+  void set_from_geo_points(const geo_point_sptr origin, const geo_point_sptr location);
+
+
+protected:
+
+  geo_point_sptr origin_;
+
+};
+
+// ----------------------------------------------------------------------------
+/** Local Cartesian Conversion Utility.
+ *
+ * Base on the NGA GeoTrans library
+ * https://earth-info.nga.mil/GandG/update/index.php?action=home
+ *
+ * This class is a cleaned up version of the LocalCartesian and Geocentric classes provided in GeoTrans
+ * This allows a user to define a local cartesian coordinate system with any origin (expressed in WGS84)
+ */
+class VITAL_EXPORT local_cartesian
+{
+public:
+  typedef Eigen::Matrix< double, 3, 1 > vector_type;
+
+  local_cartesian();
+  virtual ~local_cartesian() = default;
+
+  /**
+   * Set local origin parameters as inputs and sets the corresponding state variables.
+   *
+   *    origin[0]                : Longitude of the local origin, in degrees         (input)
+   *    origin[1]                : Latitude of the local origin, in degrees          (input)
+   *    origin[2]                : Ellipsoid height of the local origin, in meters   (input)
+   *    orientation              : Orientation angle of the local cartesian coordinate system,
+   *                               in radians                                        (input)
+   */
+  void set_origin(const vector_type& origin, double orientation=0);
+
+  /**
+   * The function convertFromGeodetic converts geodetic coordinates
+   * (latitude, longitude, and height) to local cartesian coordinates (X, Y, Z),
+   * according to the WGS84 ellipsoid and local origin parameters.
+   *
+   *    geodetic_coordinate[0]  : Geodetic longitude, in degrees                       (input)
+   *    geodetic_coordinate[1]  : Geodetic latitude, in degrees                        (input)
+   *    geodetic_coordinate[2]  : Geodetic height, in meters                           (input)
+   *    cartesian_coordinate[0] : Calculated local cartesian X coordinate, in meters   (output)
+   *    cartesian_coordinate[1] : Calculated local cartesian Y coordinate, in meters   (output)
+   *    cartesian_coordinate[2] : Calculated local cartesian Z coordinate, in meters   (output)
+   */
+  void convertFromGeodetic(const vector_type& geodetic_coordinate, vector_type& cartesian_coordinate);
```

Comment:
why are these functions camel case?

Reply:
I took all this code from geotrans
I tried preserved it as much as possible.
I can go back and format it to kwiver style if that is preferred.

Reply:
I think, unless it is intended that this is an exact copy of some other code (e.g. to enable trivial copy/paste between here and some other location to be done in the future), it would be preferred to follow KWIVER naming conventions.

Reply:
Yes, we should use KWIVER naming conventions.